core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -169,6 +175,26 @@ public static Catalog loadCatalog(
     return catalog;
   }
 
+  public static Catalog buildIcebergCatalog(String name, Map<String, String> options, Configuration conf) {
+
+    String catalogImpl = options.get(CatalogProperties.CATALOG_IMPL);
+    if (catalogImpl == null) {
+      String catalogType = options.getOrDefault(ICEBERG_CATALOG_TYPE, ICEBERG_CATALOG_TYPE_HIVE);
+      switch (catalogType.toLowerCase(Locale.ENGLISH)) {
+        case ICEBERG_CATALOG_TYPE_HIVE:
+          catalogImpl = ICEBERG_CATALOG_HIVE;
+          break;
+        case ICEBERG_CATALOG_TYPE_HADOOP:
+          catalogImpl = ICEBERG_CATALOG_HADOOP;
+          break;
+        default:
+          throw new UnsupportedOperationException("Unknown catalog type: " + catalogType);
+      }
+    }
+    return CatalogUtil.loadCatalog(catalogImpl, name, options, conf);
+
```

Comment:
Looks like this newline should be above the return to separate it from the control flow.

Reply:
done